### PR TITLE
Allow installtion without Git

### LINKED ISSUE #1205: Fails to install without "git"
With the newly released 0.12.0 (congrats!), if `git` is installed, all works fine (as expected), but if `git` is not installed, the installation fails:
```console
$ docker run -it --rm python:3.6-slim bash
root@fb6e7113d667:/# pip install hy
Collecting hy
  Downloading hy-0.12.0.tar.gz (63kB)
    100% |████████████████████████████████| 71kB 1.6MB/s 
    Complete output from command python setup.py egg_info:
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/tmp/pip-build-3xk7ny_l/hy/setup.py", line 35, in <module>
        (["git", "describe", "--tags", "--dirty"])
      File "/usr/local/lib/python3.6/subprocess.py", line 336, in check_output
        **kwargs).stdout
      File "/usr/local/lib/python3.6/subprocess.py", line 403, in run
        with Popen(*popenargs, **kwargs) as process:
      File "/usr/local/lib/python3.6/subprocess.py", line 707, in __init__
        restore_signals, start_new_session)
      File "/usr/local/lib/python3.6/subprocess.py", line 1326, in _execute_child
        raise child_exception_type(errno_num, err_msg)
    FileNotFoundError: [Errno 2] No such file or directory: 'git'
    
    ----------------------------------------
Command "python setup.py egg_info" failed with error code 1 in /tmp/pip-build-3xk7ny_l/hy/
```

This behavior appears to have been introduced in https://github.com/hylang/hy/pull/1135, which looks like it accounts for `git` failing, but not for `git` not being installed at all. :sweat_smile:

### PR DESCRIPTION
**Pull request by Kodiologist**
Fixes #1205.

If all's well, this should probably be released as 0.12.1, since #1205 is kind of nasty to leave in a release.

### CODE REVIEW

**Comment by paultag** [comment]
0.12.1 sgtm; i'll merge and push

**Comment by tianon** [comment]
:+1: :heart:

```console
$ docker run -it --rm python:3.6-slim bash
root@0da57fa95d46:/# pip install https://github.com/Kodiologist/hy/archive/version-gen-updates.tar.gz
Collecting https://github.com/Kodiologist/hy/archive/version-gen-updates.tar.gz
  Downloading https://github.com/Kodiologist/hy/archive/version-gen-updates.tar.gz
     \ 389kB 1.2MB/s
Collecting rply>=0.7.0 (from hy===unknown)
  Downloading rply-0.7.4-py2.py3-none-any.whl
Collecting astor>=0.5 (from hy===unknown)
  Downloading astor-0.5-py2.py3-none-any.whl
Collecting clint>=0.4 (from hy===unknown)
  Downloading clint-0.5.1.tar.gz
Collecting appdirs (from rply>=0.7.0->hy===unknown)
  Downloading appdirs-1.4.0-py2.py3-none-any.whl
Collecting args (from clint>=0.4->hy===unknown)
  Downloading args-0.1.0.tar.gz
Installing collected packages: appdirs, rply, astor, args, clint, hy
  Running setup.py install for args ... done
  Running setup.py install for clint ... done
  Running setup.py install for hy ... done
Successfully installed appdirs-1.4.0 args-0.1.0 astor-0.5 clint-0.5.1 hy-unknown rply-0.7.4
root@0da57fa95d46:/# hy
hy unknown using CPython(default) 3.6.0 on Linux
=> (+ 2 2)
4
```

**Comment by Kodiologist** [comment]
The first Travis check was probably triggered by me accidentally pushing the branch to hylang/hy instead of Kodiologist/hy, so it can be ignored.

File: setup.py [hunk 1]
```diff
@@ -37,8 +38,11 @@
         .replace('-', '+', 1).replace('-', '.'))
     with open(VERSIONFILE, "wt") as o:
         o.write("__version__ = {!r}\n".format(__version__))
-except subprocess.CalledProcessError:
-    __version__ = "unknown"
+except (subprocess.CalledProcessError, OSError):
+  if os.path.exists(VERSIONFILE):
```
Comment (berkerpeksag) [inline setup.py]: Indentation is inconsistent with the rest of the file.

**Comment by kirbyfan64** [comment]
WTF is going on with Travis...

**Comment by Kodiologist** [comment]
I made the same mistake again. I reset the remote for my tracking branch to `version-gen-updates`, so it shouldn't happen a third time.

**Comment by Kodiologist** [comment]
@hylang/core Somebody approve this so @paultag can merge it.

**Comment by paultag** [comment]
@Kodiologist get the travis builds green and i'll do the merge without an ack :+1:

**Comment by Kodiologist** [comment]
The failure is just because Travis tried to test a branch after I'd removed it. (I had accidentally pushed to hylang/hy instead of Kodioloigst/hy.) You can click through to read Travis's logs of the attempts.

**Comment by paultag** [comment]
Great, let's just re-run it and pass -- accept nothing on faith  :)

**Comment by gilch** [comment]
Still failing. Maybe it would work if you opened a new pull request.

**Comment by gilch** [comment]
Still failing. Maybe it would work if you opened a new pull request.

**Comment by Kodiologist** [comment]
@gilch Nope, it doesn't. See #1208.

@paultag I'm asking you to *not* accept Travis's reports on faith. Look at what commits the two Travis jobs are trying to test, look at how the fails arose, and try the PR on your own machine (in particular, see that it fixes the bug it's supposed to, which the test suite doesn't include a test for). Travis can be a nice convenience, but it's fragile. It's already broken once before in the few months I'm been working on Hy.

**Comment by gilch** [comment]
I meant a new PR with only the one branch.

**Comment by Kodiologist** [comment]
@gilch That's what I did.

**Comment by gilch** [comment]
Travis cached it? Perhaps if you made a new pull with a new branch name?

**Comment by Kodiologist** [comment]
I officially disapprove of this careless use of Travis reports, but since I seem to be in the minority here, I'll try pushing to hylang/hy, so Travis can check the branch it's trying to check.

**Comment by Kodiologist** [comment]
It didn't work (Travis didn't re-run the check), so a new PR it is.

**Pull request closed without merging**